### ISSUE TITLE
DynamoDB tables set to "on-demand" fail to deploy if they contain Secondary Indexes

### ISSUE THREAD

**Issue by Jaystified** [localstack/localstack#1737]
For example, this cloudformation template:

```json
{
  "AWSTemplateFormatVersion": "2010-09-09",
  "Transform": "AWS::Serverless-2016-10-31",
  "Resources": {
    "DynamoOnDemand": {
      "Type": "AWS::DynamoDB::Table",
      "Properties": {
        "TableName": "ondemandtable",
        "AttributeDefinitions": [
          {
            "AttributeName": "foo",
            "AttributeType": "S"
          },
          {
            "AttributeName": "bar",
            "AttributeType": "S"
          },
          {
            "AttributeName": "baz",
            "AttributeType": "S"
          }
        ],
        "KeySchema": [
          {
            "AttributeName": "foo",
            "KeyType": "HASH"
          },
          {
            "AttributeName": "bar",
            "KeyType": "RANGE"
          }
        ],
        "GlobalSecondaryIndexes": [
          {
            "IndexName": "by.baz",
            "KeySchema": [
              {
                "AttributeName": "foo",
                "KeyType": "HASH"
              },
              {
                "AttributeName": "baz",
                "KeyType": "RANGE"
              }
            ],
            "Projection": {
              "ProjectionType": "ALL"
            }
          }
        ],
        "BillingMode": "PAY_PER_REQUEST"
      }
    }
  }
}
```

Fails to deploy due to an error: `botocore.exceptions.ClientError: An error occurred (ValidationException) when calling the CreateTable operation: No provisioned throughput specified for the global secondary index`

But, secondary indexes in `"BillingMode": "PAY_PER_REQUEST"` tables should not contain provisioning settings (and deployment on AWS will fail if they do)

Related? #1089

**Comment by whummer** [comment]
Looks like a duplicate of #1193

**Comment by mechanicalpete** [comment]
I'm experiencing this as well with the latest version: 0.11.1

I have a work around where I uncomment the provisioning to provide values. Almost a deal breaker as this is also the production CloudFormation template so I need to make sure to re-comment-out before pushing changes.

**Comment by marcindulak** [comment]
Issue still present in https://github.com/localstack/localstack/tree/v0.11.2

**Comment by rikbruil** [comment]
@mechanicalpete My workaround does a similar thing, but I have a script that copies my 'real' cloudformation template file, which I update in the script using the yq tool to add the throughput settings in my case.

Then the script does a local deploy using my edited file rather than the real production file and removes the local file after deployment again. I also added the filename of the local only template to gitignore to prevent us from committing it.

**Comment by whummer** [comment]
Thanks for the update @rikbruil @marcindulak @mechanicalpete . This should be fixed in #2844 - can you please pull the Docker image and give it another try with the `latest` version? Please report here if the problem persists. Thanks
